wdl-lint/src/rules/command_shellcheck.rs

```diff
@@ -0,0 +1,357 @@
+const SHELLCHECK_SUPPRESS: &[&str] = &[
+    "1009", // the mentioned parser error was in...
+    "1072", // Unexpected..
+    "1083", // this {/} is literal
+    "2043", // this loop will only ever run once for constant value
+    "2050", // This expression is constant
+    "2157", // Argument to -n is always true due to literal strings
+    "2193", // The arguments to this comparison can never be equal
+];
+
+/// The identifier for the command section ShellCheck rule.
+const ID: &str = "CommandSectionShellCheck";
+
+/// A ShellCheck comment
+///
+/// The file and fix fields are ommitted as we have no use for them.
+#[derive(Clone, Debug, Deserialize)]
+struct ShellCheckComment {
+    /// line number comment starts on
+    pub line: usize,
+    /// line number comment ends on
+    #[serde(rename = "endLine")]
+    pub end_line: usize,
+    /// column comment starts on
+    pub column: usize,
+    /// column comment ends on
+    #[serde(rename = "endColumn")]
+    pub end_column: usize,
+    /// severity of the comment
+    pub level: String,
+    /// shellcheck error code
+    pub code: usize,
+    /// message associated with the comment
+    pub message: String,
+}
+
+/// Run shellcheck
+///
+/// writes command text to stdin of shellcheck process
+/// and returns parsed ShellCheckComments
+fn run_shellcheck(command: &str) -> Result<Vec<ShellCheckComment>> {
+    let mut sc_proc = process::Command::new(SHELLCHECK_BIN)
+        .args([
+            "-s",
+            "bash",
+            "-f",
+            "json",
+            "-e",
+            &SHELLCHECK_SUPPRESS.join(","),
+            "-",
+        ])
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()
+        .context("error calling shellcheck")?;
+    {
+        let mut proc_stdin = sc_proc
+            .stdin
+            .take()
+            .context("error obtaining stdin of shellcheck")?;
+        proc_stdin.write_all(command.as_bytes())?;
+    }
+
+    let output = sc_proc
+        .wait_with_output()
+        .context("error calling shellcheck")?;
+
+    // shellcheck returns exit code 1 if
+    // any checked files result in comments
+    // so cannot check with status.success()
+    match output.status.code() {
+        Some(0) | Some(1) => serde_json::from_slice::<Vec<ShellCheckComment>>(&output.stdout)
+            .context("failed to parse shellcheck output"),
+        Some(code) => bail!("error calling shellcheck: exit code {}", code),
+        None => bail!("error calling check: process interrupted"),
+    }
+}
+
+/// Runs ShellCheck on a command section and reports violations
+#[derive(Default, Debug, Clone, Copy)]
+pub struct CommandSectionShellCheck;
+
+impl Rule for CommandSectionShellCheck {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that command blocks are free of ShellCheck violations."
+    }
+
+    fn explanation(&self) -> &'static str {
```

Comment:
Perhaps provide a link to ShellCheck in parenthesis after the first introduction of the name.